homeassistant/components/plex/sensor.py

```diff
@@ -3,76 +3,31 @@
 import logging
 import plexapi.exceptions
 import requests.exceptions
-import voluptuous as vol
-
-from homeassistant.components.switch import PLATFORM_SCHEMA
-from homeassistant.const import (
-    CONF_NAME,
-    CONF_HOST,
-    CONF_PORT,
-    CONF_TOKEN,
-    CONF_SSL,
-    CONF_URL,
-    CONF_VERIFY_SSL,
-)
+
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
-import homeassistant.helpers.config_validation as cv
 
-from .const import DEFAULT_HOST, DEFAULT_PORT, DEFAULT_SSL, DEFAULT_VERIFY_SSL
-from .server import PlexServer
+from .const import DOMAIN as PLEX_DOMAIN, SHARED_SERVER
 
 DEFAULT_NAME = "Plex"
 _LOGGER = logging.getLogger(__name__)
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_TOKEN): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
-        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
-    }
-)
-
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Plex sensor."""
-    name = config.get(CONF_NAME)
-    plex_host = config.get(CONF_HOST)
-    plex_port = config.get(CONF_PORT)
-    plex_token = config.get(CONF_TOKEN)
-    verify_ssl = config.get(CONF_VERIFY_SSL)
-
-    plex_url = "{}://{}:{}".format(
-        "https" if config.get(CONF_SSL) else "http", plex_host, plex_port
-    )
-
-    try:
-        plex_server = PlexServer(
-            {CONF_URL: plex_url, CONF_TOKEN: plex_token, CONF_VERIFY_SSL: verify_ssl}
-        )
-        plex_server.connect()
-    except (
-        plexapi.exceptions.BadRequest,
-        plexapi.exceptions.Unauthorized,
-        plexapi.exceptions.NotFound,
-    ) as error:
-        _LOGGER.error(error)
-        return
-
-    add_entities([PlexSensor(name, plex_server)], True)
+    server_id = hass.data[PLEX_DOMAIN][SHARED_SERVER]
+    plexserver = hass.data[PLEX_DOMAIN][server_id]
+    add_entities([PlexSensor(plexserver)], True)
 
 
 class PlexSensor(Entity):
     """Representation of a Plex now playing sensor."""
 
-    def __init__(self, name, plex_server):
+    def __init__(self, plex_server):
         """Initialize the sensor."""
-        self._name = name
+        self._name = DEFAULT_NAME
```

Comment:
Can we use the `machineIdentifier` for `unique_id` property as we do in the media_player platform? Then we would allow users to still rename the entity.

Reply:
Done. Once multiple sensors are allowed, a better default name will also be provided.